implementations/rust/ockam/ockam_command/src/node/delete.rs

```diff
@@ -40,32 +41,94 @@ impl DeleteCommand {
     }
 }
 
+enum DeleteMode {
+    All,
+    Selected(Option<Vec<String>>),
+    Single(Option<String>),
```

Comment:
Perhaps, instead of using `Option` in these two variants, you could add a new variant called `None`, so that you don't have to check the `Option` when deleting the nodes. It would also help removing the `unwrap` from line 96: `let selected_node_names = option_selected_node_names.unwrap();`

To make things even easier to understand, I would add a `Default` variant that should be used when `DeleteCommand`'s `node_name` is `None`.